File: Python/Sorting/insertion_sort.py
# #taking input array from user
# num_array = list()
# num = raw_input("Enter how many elements you want:")
# print 'Enter numbers in array: '
# for i in range(int(num)):
#     n = raw_input("num :")
#     num_array.append(int(n))
# print 'ARRAY: ',num_array

# #reading input array from file
# with open('data.txt', 'r') as myfile:
#     data=myfile.read().replace('\n', '')
#     num_array = data.split(",")


#Insertion sort
def insertion(arr):
	for i in range(1,len(arr)):
		key = arr[i]
		j = i-1
		while j>=0 and arr[j]>key:
			arr[j+1] = arr[j]
			j = j-1
		arr[j+1] = key
	print 'Sorted: ',arr


if __name__ == '__main__':
	arr = [7,4,9,6,5,3,19,37,1,6,9,5]
	print 'Original: ',arr
	insertion(arr)

File: Java/Sorting/InsertionSort.java
package soritng;
import java.util.Arrays;

public class InsertionSort {
	public static void main(String[] args) {
		int[] arr = {7,4,9,6,5,3,19,37,1,7,4,6,4};
		System.out.println("Original Array: "+Arrays.toString(arr));

		int[] sorted_arr = insertion_sort(arr);
		System.out.println("Sorted Array: "+Arrays.toString(sorted_arr));
	}

	public static int[] insertion_sort(int[] arr) {
		int key = 0,j=0;
		for ( int i=1;i<arr.length;i++ ) {
			key = arr[i];
			j = i-1;
			while ( j>=0 && arr[j]>key ) {
				arr[j+1] = arr[j];
				j = j-1;
			}
			arr[j+1] = key;
		}

		return arr;
	}
}